PWGGA/GammaConvBase/AliConversionPhotonCuts.cxx

```diff
@@ -4779,6 +4779,63 @@ Bool_t AliConversionPhotonCuts::RejectToCloseV0s(AliAODConversionPhoton* photon,
   return kTRUE;
 }
 
+///________________________________________________________________________
+Bool_t AliConversionPhotonCuts::AllowedBySharedElectronCut(set<Int_t> &theLabels, AliAODConversionPhoton &thePhoton) const {
+  // todo: check if two separate sets for electrons and positrons are safe to use. I think it is and would bring speedup
+
+  // since it is quite rare that this cut cuts, we insert the positive track before checking if the negative one can be inserted as well
+  auto lItInsWorkedPositive = theLabels.insert(thePhoton.GetTrackLabelPositive());
+  if (!lItInsWorkedPositive.second){
+    return  kFALSE;
+  }
+
+  Bool_t lInsWorkedNegative = theLabels.insert(thePhoton.GetTrackLabelNegative()).second;
+  if (!lInsWorkedNegative){
+    theLabels.erase(lItInsWorkedPositive.first);
+    return kFALSE;
```

Comment:
we may want to check the chi2 values instead of simply accepting the first photon candidate in the list - to be discussed